### PR TITLE
[Jest] Migrate `style.test.js`

### PR DESCRIPTION
**Pull request by wipfli**
- Move and rename
- Run jscodeshift
- Run custom codemods and lint
- Manual edits

## Launch Checklist

This pull request continues the work of @astridx's #733. I added 1 commit (see https://github.com/wipfli/maplibre-gl-js/commit/a26b7a17a931e5355d9b73040daa81b9bbcf10f1) to make the tests pass.
Then I rebased the different manual commits after the automatic ones for easier review. Thanks Astrid for contributing the migration of this test!

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: src/style/style.test.ts [hunk 1]
```diff
@@ -1965,72 +2032,78 @@ describe('Style#query*Features', () => {
...
-        const SourceType = function () {};
+    test('adds factory function', done => {
+        const style = new Style(getStubMap());
+        const SourceType = function () {} as any as SourceClass;
 
         // expect no call to load worker source
         style.dispatcher.broadcast = function (type) {
             if (type === 'loadWorkerSource') {
-                t.fail();
+                done.fail();
```
Comment (HarelM) [inline src/style/style.test.ts:2061]: We said this doesn't work, right?

Comment (wipfli) [reply src/style/style.test.ts]: Yes. Fixed.

**Comment by astridx** [comment]
Thank you for taking over my branch. 

When I uploaded it, I was sure about the enabled code parts. At the moment I am unsure in parts. 

In the test "validates filter by default" it should be "havebeencalledtimes(1)". (line 1645) 

And while correcting I noticed that I forgot done in line 101 (test "loads plugin immediately if already registered").  

That's where I was currently stuck in my brunch. 

I'm sorry for making a mess here again.

**Review by HarelM — CHANGES_REQUESTED** [review]
Very hard to review due to compare misalignment... :-(

**Comment by wipfli** [comment]
#764

**Pull request closed without merging**